m2c/types.py

```diff
@@ -503,23 +526,61 @@ def get_field(
+            if (
+                data.struct.is_union
+                and data.struct.typepool is not None
+                and data.struct.typepool.union_field_overrides
+            ):
+                # Try to get the struct name (prefer typedef_name, then tag_name)
+                struct_name = data.struct.typedef_name or data.struct.tag_name
+                if struct_name and struct_name in data.struct.typepool.union_field_overrides:
+                    preferred_field_name = data.struct.typepool.union_field_overrides[struct_name]
+                    has_union_override = True
+                    # Reorder possible_fields to put the preferred field first
+                    for i, field in enumerate(possible_fields):
+                        if field.name == preferred_field_name:
+                            # Move this field to the front
+                            possible_fields.insert(0, possible_fields.pop(i))
+                            break
+
             # Recursively check each of the `possible_fields` to find the best matches
             # for fields at the given offset.
-            for field in possible_fields:
+            for i, field in enumerate(possible_fields):
                 inner_offset_bits = offset - field.offset
+                # When we have a union override and this is the preferred field (first in the
+                # reordered list), be very lenient: try with exact match first, but fall back
+                # to allowing size mismatches if needed
+                is_override_field = has_union_override and i == 0
                 subpath, subtype, sub_remaining_offset = field.type.get_field(
                     inner_offset_bits,
                     target_size=target_size,
-                    exact=exact,
+                    exact=False if is_override_field else exact,
                 )
+                # If override field didn't match with target_size, try without it
+                if subpath is None and is_override_field:
+                    subpath, subtype, sub_remaining_offset = field.type.get_field(
+                        inner_offset_bits,
+                        target_size=None,
+                        exact=False,
+                    )
                 if subpath is None:
                     continue
                 subpath.insert(0, field.name)
                 possible_results.append((subpath, subtype, sub_remaining_offset))
 
+                # If we have a union override and this is the first field (which we reordered
+                # to be the preferred field), return it immediately regardless of size match.
+                if is_override_field and sub_remaining_offset == 0:
+                    return possible_results[-1]
+
                 # Check if this subfield is an exact match. If it is, return it.
+                # However, if we have a union override, we skip this check because we want
+                # to return the overridden field above.
```

Comment:
I don't get this part. Even if we skip this check, won't we return the first thing with sub_remaining_offset == 0 right after the loop anyway?